odoo/addons/base/models/ir_cron.py

```diff
@@ -533,6 +615,33 @@ def _notifydb(self):
             cr.execute(query, [self.env.cr.dbname])
         _logger.debug("cron workers notified")
 
+    def _add_progress(self, *, timed_out=0):
+        """ Create a progress record for the given cron. """
+        progress = self.env['ir.cron.progress'].create([{
+                'cron_id': self.id,
+                'remaining': 0,
+                'done': 0,
+                'timed_out': timed_out,
+        }])
+        return self.with_context(ir_cron_progress_id=progress.id)
```

Comment:
suggestion on top of my head
```suggestion
    def _add_progress(self, *, last_timed_out=0):
        """ Create a progress record for the given cron. """
        progress = self.env['ir.cron.progress'].create([{
                'cron_id': self.id,
                'remaining': 0,
                'done': 0,
                'timed_out': last_timed_out + 1,
        }])
        return self.with_context(ir_cron_progress_id=progress.id)
```
and we change the few places where we call it. I don't think there'll be ever a case where we want to create a progress that is *not* timed out by default.

Reply:
Also, missing docstring for that parameter.